python/lib/dependabot/python/update_checker/pip_compile_version_resolver.rb

```diff
@@ -80,9 +80,13 @@ def fetch_latest_resolvable_version_string(requirement:)
                   run_pip_compile_command(
                     "pyenv exec pip-compile -v #{pip_compile_options(filename)} -P #{dependency.name} #{filename}"
                   )
-                  # Run pip-compile a second time, without an update argument,
-                  # to ensure it handles markers correctly
-                  write_original_manifest_files unless dependency.top_level?
+
+                  next if dependency.top_level?
+
+                  # Run pip-compile a second time for transient dependencies
+                  # to make sure we do not update dependencies that are
+                  # superfluous.
+                  write_original_manifest_files
                   run_pip_compile_command(
                     "pyenv exec pip-compile #{pip_compile_options(filename)} #{filename}"
                   )
```

Comment:
Based on your comment in https://github.com/dependabot/dependabot-core/issues/5903#issuecomment-1282101954 it looks like we should also be able to get rid of this second call to `pip-compile`. Is there a reason we keep it here but remove it from the file updater?

Reply:
If we remove it, Dependabot will create PR's for dependencies that are superfluous when the config option `dependency-type` is set to `indirect`. It's a minor inconvenience imho but I think it's better to keep the output/results of Dependabot the same as before.

We could remove it from the file updater because it appears that the reason it ran pip-compile twice in both files had a different reasons. When I removed it, it had no effect on any tests nor on the output of any of my manual tests. There is already logic to copy the old header text of the generated requirements file so there isn't really anything to "correct" it seems. It could also be that this was implemented to work around some specific behavior of pip-compile that has since been fixed.

Reply:
Thanks for the explanation! 

Here's a guess at what's happening: in the second run, because `pip-compile` doesn't use the `-P {dependency}` flag, it's able to consider all dependencies and see that there are superfluous ones after the update. With the `-P` flag pip can't know whether other dependencies are using them.

Reply:
Yes that's exactly what happens :)

Reply:
It also makes sense to me, and agree on including that in the comment 👍. Can you amend the commit to improve the explanation and rebase the PR?

Reply:
@deivid-rodriguez done!